tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -57,38 +57,41 @@ class TlaCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter
     if (state.isEmpty) {
       pretty.write(ValEx(TlaBool(true)))
     } else {
-      state.toList.sortBy(_._1).foreach{
-        case (name,value) =>
-          writer.print(s"/\\ $name = " )
-          pretty.write(value)
-          writer.println
+      state.toList.sortBy(_._1).foreach { case (name, value) =>
+        writer.print(s"/\\ $name = ")
+        pretty.write(value)
+        writer.println
       }
     }
   }
 
-  override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState] ) : Unit = {
+  override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
     val pretty = new PrettyWriter(writer)
 
     writer.println("%s MODULE counterexample %s\n".format("-" * 25, "-" * 25))
     writer.println("EXTENDS %s\n".format(rootModule.name))
 
     states.zipWithIndex.foreach {
+      case (state, 0) =>
+        writer.println("(* Constant initialization state *)")
+        writer.println(s"\nStateConst ==")
+        printStateFormula(pretty, state._2)
+        writer.println()
       case (state, j) =>
-        val i = j+1 // start counting from 1, for compatibility with TLC counterexamples
-        if(i == 1) {
+        // start counting from 0, for compatibility with TLC counterexamples
```

Comment:
This comment does not help much. Shall we say that index 0 is reserved for `ConstInit`, whereas indices `1..n` are reserved for states. However, users expect `State0` to be the initial state, so we count states from 0.

Reply:
Sure